layers/core_checks/cc_ray_tracing.cpp

```diff
@@ -530,25 +531,44 @@ bool CoreChecks::ValidateAccelerationBuffers(uint32_t info_index, const VkAccele
     }
 
     if (geom_accessor) {
-        for (uint32_t geom_index = 0; geom_index < geometry_count; ++geom_index) {
-            const Location geom_loc = loc.dot(Field::pGeometries, geom_index);
-            const auto &geom_data = geom_accessor(geom_index);
+        const Location pp_build_range_info_loc(loc.function, Field::ppBuildRangeInfos, info_i);
+        for (uint32_t geom_i = 0; geom_i < geometry_count; ++geom_i) {
+            const Location geom_loc = loc.dot(Field::pGeometries, geom_i);
+            const auto &geom_data = geom_accessor(geom_i);
             switch (geom_data.geometryType) {
                 case VK_GEOMETRY_TYPE_TRIANGLES_KHR:  // == VK_GEOMETRY_TYPE_TRIANGLES_NV
-                    skip |= buffer_check(geom_index, geom_data.geometry.triangles.vertexData,
-                                         geom_loc.dot(Field::geometry).dot(Field::triangles).dot(Field::vertexData));
-                    skip |= buffer_check(geom_index, geom_data.geometry.triangles.indexData,
-                                         geom_loc.dot(Field::geometry).dot(Field::triangles).dot(Field::indexData));
-                    skip |= buffer_check(geom_index, geom_data.geometry.triangles.transformData,
-                                         geom_loc.dot(Field::geometry).dot(Field::triangles).dot(Field::transformData));
+                {
+                    skip |= buffer_check(geom_i, geom_data.geometry.triangles.vertexData,
```

Comment:
since we already have `geom_loc` and this currently has to re-build the Location object 3 times to call down

```suggestion
                    const Location triangles_loc = geom_loc.dot(Field::triangles);
                    skip |= buffer_check(geom_i, geom_data.geometry.triangles.vertexData, triangles_loc.dot(Field::vertexData)
```

Reply:
yeah better